script/Script.s.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.25;
+
+import {IERC20} from '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+import {IBPool} from 'contracts/BPool.sol';
+import {IFaucet} from 'interfaces/IFaucet.sol';
+
+import {Script} from 'forge-std/Script.sol';
+import {Registry} from 'script/Registry.s.sol';
+
+/// @notice This base script is shared across `yarn script:{mainnet|testnet}`
+abstract contract BaseScript is Registry, Script {
+  constructor() Registry(block.chainid) {}
+}
+
+/// @notice This script will be executed by `yarn script:mainnet`
+contract MainnetScript is BaseScript {
+  function run() public {
+    assert(block.chainid == 1);
+    vm.startBroadcast();
+
+    // script logic here
+
+    vm.stopBroadcast();
+  }
+}
+
+/// @notice This script will be executed by `yarn script:testnet`
+contract TestnetScript is BaseScript {
```

Comment:
I think it'd be clearer for the scripts to be named `testnetSetup` and `mainnetSetup` (or something more a bit more descriptive of what they do)

and their npm script names could be `setup:testnet`, `setup:mainnet`

Reply:
but is not about "setting up" is about a template for a developer to do whatever